stdlib/task/src/main/java/org/ballerinalang/stdlib/task/objects/AbstractTask.java

```diff
@@ -141,66 +141,44 @@ private void validateMaxRuns(long maxRuns) throws SchedulingException {
     }
 
     /**
-     * Stops the scheduled Appointment.
-     *
-     * @param taskId ID of the task which should be stopped.
-     * @throws SchedulingException if failed to stop the task.
+     * {@inheritDoc}
      */
-    public void stop(String taskId) throws SchedulingException {
-        if (quartzJobs.containsKey(taskId)) {
-            try {
-                scheduler.deleteJob(quartzJobs.get(taskId));
-            } catch (SchedulerException e) {
-                throw new SchedulingException("Cannot cancel the task. ");
+    public void stop() throws SchedulingException {
+        try {
+            this.scheduler.shutdown(true);
+        } catch (SchedulerException e) {
+            if (log.isDebugEnabled()) {
```

Comment:
This is not a correct way. Either we have to log it here or throw an error. Shall we add the exception instance to newly created SchedulingException and handle it in upper layer?

Reply:
Done